core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+            nextMigration(rateLimit = isBlocked || activeFullMigrations >= maxAttempts)
+
+          _ <- OptionT.liftF[ReadWriteAction, Unit] {
+            workspaceQuery.withWorkspaceId(workspaceId).lock.ignore
+          }
+          _ <- OptionT.liftF(dataAccess.multiregionalBucketMigrationQuery.update(id, startedCol, Some(now)))
+        } yield (id, workspaceName)
+      }
+      _ <- getLogger[MigrateAction].info(
+        stringify(
+          "migrationId" -> id,
+          "workspace" -> workspaceName,
+          "started" -> now
+        )
+      )
+    } yield ()
+
+  final def removeWorkspaceBucketIam: MigrateAction[Unit] =
+    withMigration(_.multiregionalBucketMigrationQuery.removeWorkspaceBucketIamCondition) { (migration, workspace) =>
+      val makeError = (message: String, data: Map[String, Any]) =>
+        WorkspaceMigrationException(
+          message = s"The workspace migration failed while removing workspace bucket IAM: $message.",
+          data = Map(
+            "migrationId" -> migration.id,
+            "workspace" -> workspace.toWorkspaceName
+          ) ++ data
+        )
+
+      for {
+        (storageService, userInfo, googleProjectToBill) <- asks { d =>
+          (d.storageService, d.userInfo, d.googleProjectToBill)
+        }
+
+        _ <- workspace.errorMessage.traverse_ { message =>
+          raiseError {
+            makeError("an error exists on workspace", Map("errorMessage" -> message))
+          }
+        }
+
+        workspaceBillingAccount = workspace.currentBillingAccountOnGoogleProject.getOrElse(
+          throw makeError("no billing account on workspace", Map.empty)
+        )
+
+        // Safe to assume that this exists if the workspace exists
+        billingProject <- inTransactionT { dataAccess =>
+          OptionT {
+            dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspace.namespace))
+          }
+        }
+
+        billingProjectBillingAccount = billingProject.billingAccount.getOrElse(
+          throw makeError("no billing account on billing project", Map("billingProject" -> billingProject.projectName))
+        )
+
+        _ <- raiseWhen(billingProject.invalidBillingAccount) {
+          makeError(
+            "invalid billing account on billing project",
+            Map(
+              "billingProject" -> billingProject.projectName,
+              "billingProjectBillingAccount" -> billingProjectBillingAccount
+            )
+          )
+        }
+
+        _ <- raiseWhen(workspaceBillingAccount != billingProjectBillingAccount) {
+          makeError(
+            "billing account on workspace differs from billing account on billing project",
+            Map(
+              "workspaceBillingAccount" -> workspaceBillingAccount,
+              "billingProject" -> billingProject.projectName,
+              "billingProjectBillingAccount" -> billingProjectBillingAccount
+            )
+          )
+        }
```

Comment:
These checks were all moved from `configureGoogleProject`. We still need to check that billing information is up to date before attempting to migrate a workspace's bucket, but we're not configuring the Google project, so I moved the check here

Reply:
I think this makes sense. Is the strictly needed to migrate a bucket or is it more of a Good Clean Living™ thing?

Reply:
We do need a valid billing account on the workspace so that we can create buckets and perform other GCP operations. The billing project billing account isn't such a big deal and we have some checks in place now that make it safer to assume that the billing project's and workspace's billing accounts are the same. I still felt there wasn't much downside to keeping these extra checks in place for the sake of Good Clean Living™